Doc/glossary.rst

```diff
@@ -1281,6 +1313,16 @@ Glossary
       See also :term:`binary file` for a file object able to read and write
       :term:`bytes-like objects <bytes-like object>`.
 
+   thread state
+      In Python's C API, a thread state is a structure that holds
+      information about the current thread, typically in the :term:`current thread state`
+      pointer. A thread state can be attached or detached. An :term:`attached thread state`
+      is required to call most of the C API, unless a function explicitly documents
+      otherwise.
+
+      See :ref:`Thread State and the Global Interpreter Lock <threads>` for more
+      information.
```

Comment:
```suggestion
   runtime context
      (AKA execution context)
      The information used by the Python runtime to run in an OS thread.
      This includes the current exception, if any, and the state of the bytecode interpreter.
      In Python's C API, it is known as a :term:`thread state`.
      
      Each context is bound to a single OS thread, but each thread may have
      many contexts.  At most one of them may be active at once, known as the
      :term:`current runtime context`.  The current context may be attached
      or detached.  An :term:`attached runtime context` is required to call most
      of the C API, unless a function explicitly documents otherwise.  Python code
      only runs in an attached context.
      
      Each context belongs to a single interpreter, but each interpreter may have
      many contexts.  It may even have multiple contexts for the same OS thread.
      Contexts from multiple interpreters may be bound to the same thread,
      but only one can be active in that thread at any given moment.

      See :ref:`Thread State and the Global Interpreter Lock <threads>` for more
      information.
      
   thread state
      The concrete C API implementation of a :term:`runtime context`.
      See: :c:data:`PyThreadState`.
```